VirtualFleetDocker.py

```diff
@@ -0,0 +1,257 @@
+import argparse
+import json
+import logging
+import os
+import pathlib
+import shutil
+import signal
+import socket
+import sys
+import time
+from concurrent.futures import ThreadPoolExecutor
+
+import docker
+
+LOG_LAST_LINES = 5
+CONFIG_PATH = "./config"
+TMP_CONFIG_PATH = os.path.join(CONFIG_PATH, "tmp-configs")
+EXTERNAL_SERVER_CONFIG = os.path.join(CONFIG_PATH, "external-server/config.json")
+MODULE_GATEWAY_CONFIG = os.path.join(CONFIG_PATH, "module-gateway/config.json")
+VERNEMQ_CONFIG = os.path.join(CONFIG_PATH, "vernemq")
+LOGS_DIR = "logs"
+runningContainers = []
+
+class FileDoesntExistException(Exception):
+    pass
+
+class PortOutOfRangeException(Exception):
+    def __init__(self, message="Ports out of range, it has to be: 1024 < my port < 65535"):
+        super().__init__(message)
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", default=os.path.join(CONFIG_PATH, "virtual-fleet-config.json"), help="path to json configuration file")
+    return parser.parse_args()
+
+def check_paths(arguments):
+    if not os.path.exists(arguments.config):
+        raise FileDoesntExistException(f"Json file {arguments.config} doesn't exist")
+
+def is_port_available(port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(("localhost", port)) != 0
+
+def check_vehicles_uniqueness(vehicles):
+    vehicle_names = set()
+    for vehicle in vehicles:
+        vehicle_id = f"{vehicle['company']}-{vehicle['name']}"
+        if vehicle_id in vehicle_names:
+            raise ValueError(f"Vehicle {vehicle_id} is not unique")
+        vehicle_names.add(vehicle_id)
+
+def create_config_files(vehicle):
+    with open(os.path.abspath(EXTERNAL_SERVER_CONFIG), "r") as file:
+        config = json.load(file)
+    config["company_name"] = vehicle["company"]
+    config["car_name"] = vehicle["name"]
+    tmp_config_address = os.path.join(TMP_CONFIG_PATH, f"{vehicle['company']}-{vehicle['name']}-external-server-config.json")
```

Comment:
External servers and module gateways currently restrict parameter changes via flags, necessitating the generation of a temporary configuration (tmp_config) for each container. However, this issue may be resolved in the future by enabling this capability within the external server and module gateway.

Reply:
Sorry, I was unclear, the variable name tmp_config_address could be path tmp_config_path. The Constatnt TMP_CONFIG_PATH could be TMP_CONFIG_DIR, since its directory for all tmp configs